build.zig

```diff
@@ -9,16 +9,26 @@ const version = "0.1.2"; // managed by release.sh
 /// Adds the required packages to the given executable.
 ///
 /// This is used for providing the dependencies for main executable as well as the tests.
-fn addPackages(allocator: std.mem.Allocator, exe: *std.build.LibExeObjStep) !void {
-    exe.addPackagePath("clap", "libs/zig-clap/clap.zig");
-    for ([_][]const u8{ "file", "gen", "wav" }) |package| {
-        const path = try std.fmt.allocPrint(allocator, "src/{s}.zig", .{package});
-        defer allocator.free(path);
-        exe.addPackagePath(package, path);
-    }
+fn addPackages(b: *std.Build, exe: *std.build.LibExeObjStep) !void {
+    exe.addModule("clap", b.createModule(.{
+        .source_file = .{ .path = "libs/zig-clap/clap.zig" },
+        .dependencies = &.{},
+    }));
+    exe.addModule("file", b.createModule(.{
+        .source_file = .{ .path = "src/file.zig" },
+        .dependencies = &.{},
+    }));
+    exe.addModule("gen", b.createModule(.{
+        .source_file = .{ .path = "src/gen.zig" },
+        .dependencies = &.{},
+    }));
+    exe.addModule("wav", b.createModule(.{
+        .source_file = .{ .path = "src/wav.zig" },
+        .dependencies = &.{},
+    }));
 }
 
-pub fn build(b: *std.build.Builder) !void {
+pub fn build(b: *std.Build) !void {
     // Create an allocator.
     var gpa = std.heap.GeneralPurposeAllocator(.{}){};
     const allocator = gpa.allocator();
```

Comment:
why not using `b.allocator`?

Reply:
applied to main in 1b84f60e203600078d9a778979f540d5ade57019